staging/src/k8s.io/component-base/metrics/prometheus/restclient/metrics.go

```diff
@@ -54,13 +54,53 @@ var (
 		},
 		[]string{"code", "method", "host"},
 	)
+
+	execPluginCertExpiration = k8smetrics.NewGauge(
+		&k8smetrics.GaugeOpts{
+			Name: "rest_client_exec_plugin_certificate_expiration_seconds",
+			Help: "Gauge of the nearest expiration of the client " +
+				"certificate(s) managed by the auth exec plugin. The value " +
+				"is in seconds since January 1, 1970 UTC. If auth exec " +
+				"plugins are unused or manage no TLS certificates, the " +
+				"value will be 0.",
+		},
+	)
+
+	execPluginCertRotation = k8smetrics.NewHistogram(
+		&k8smetrics.HistogramOpts{
+			Name: "rest_client_exec_plugin_certificate_rotation_age",
+			Help: "Histogram of the number of seconds the last auth exec " +
+				"plugin client certificate lived before being rotated. " +
+				"If auth exec plugin client certificates are unused, " +
+				"histogram will contain no data.",
+			Buckets: []float64{
```

Comment:
what was the rationale on these buckets?
* what is the 1 minute bucket intended to capture?
* do we need something between 1 day and 1 week?
* do we need something between 1 year and 4 years?

Reply:
My thinking is the values are exponentially increasing to capture a wide range:

- **1 minute, 1 hour buckets**: capture rotations happening probably way too quickly, indicates a problem.
- **4 hours - 1 month buckets**: captures what I think is a healthy range of rotation cadences
- **3 months - 4 years buckets**: captures rotation cadences are that too long or WAY too long. 

> what is the 1 minute bucket intended to capture?

Way too frequent rotations.

> do we need something between 1 day and 1 week?

Probably not but if you feel strongly I can add it. Its not that significantly different than 1 day in my mind.

> do we need something between 1 year and 4 years?

Again I don't think so but if you feel strongly I can add it. I figure 1+ year rotation cadence is way too slow no need to capture a range of _way too slow_ rotations. I added 4 years to get a picture of the outlier certs that finally rotated after a super long time.

Reply:
I don't feel that strongly. If @brancz / @logicalhan are happy with these, and we can clearly describe the intent of the buckets (which we should in the docs around these metrics), that's fine

Reply:
The 1 min bucket I'm not too sure about, but since there are no other dimensions on this histogram, I don't feel strongly about it either. Remember that it's cumulative buckets, so there will only be an increment in that bucket if it's between 0s and 60s, I doubt we'll ever see an observation like that, I feel something between 10min-30min would be more useful and realistic here.

Reply:
Changed 1m to 10m and 30m. I also added a descriptive comment.

Reply:
Changed the 1m into a 10m and 30m bucket. Also added a descriptive comment.